File: app/src/main/java/org/kiwix/kiwixmobile/base/BaseFragment.java
package org.kiwix.kiwixmobile.base;

import android.app.Activity;
import android.content.Context;
import android.os.Build;

import org.kiwix.kiwixmobile.KiwixApplication;

import androidx.fragment.app.Fragment;

/**
 * All fragments should inherit from this fragment.
 */

public abstract class BaseFragment extends Fragment {

  @Override
  public void onAttach(Context context) {
    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
      KiwixApplication.getApplicationComponent().inject(this);
    }
    super.onAttach(context);
  }

  @SuppressWarnings("deprecation")
  @Override
  public void onAttach(Activity activity) {
    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
      KiwixApplication.getApplicationComponent().inject(this);
    }
    super.onAttach(activity);
  }
}

File: app/src/main/java/org/kiwix/kiwixmobile/base/BaseFragment.kt
package org.kiwix.kiwixmobile.base

import android.content.Context
import androidx.fragment.app.Fragment
import org.kiwix.kiwixmobile.KiwixApplication
import org.kiwix.kiwixmobile.di.components.ActivityComponent

/**
 * All fragments should inherit from this fragment.
 */

abstract class BaseFragment : Fragment() {

  override fun onAttach(context: Context?) {
    super.onAttach(context)
    inject(
      KiwixApplication.getApplicationComponent().activityComponent()
        .activity(activity!!)
        .build()
    )
  }

  abstract fun inject(activityComponent: ActivityComponent)
}
